server/src/dependency/in_memory.rs

```diff
@@ -1,94 +1,157 @@
 use {
     crate::dependency::shared::*,
-    moved_app::{Application, CommandActor},
+    moved_app::{Application, ApplicationReader, CommandActor},
     moved_genesis::config::GenesisConfig,
+    std::sync::Arc,
 };
 
 pub type Dependency = InMemoryDependencies;
 
-pub fn create(genesis_config: &GenesisConfig) -> Application<InMemoryDependencies> {
-    Application::new(InMemoryDependencies, genesis_config)
+pub fn create(
+    genesis_config: &GenesisConfig,
+) -> (
+    Application<InMemoryDependencies>,
+    ApplicationReader<InMemoryDependencies>,
+) {
+    let deps = InMemoryDependencies::new();
+    let reader_deps = deps.reader();
+
+    (
+        Application::new(deps, genesis_config),
+        ApplicationReader::new(reader_deps, genesis_config),
+    )
 }
 
-pub struct InMemoryDependencies;
+pub struct InMemoryDependencies {
+    memory_reader: moved_blockchain::in_memory::SharedMemoryReader,
+    memory: Option<moved_blockchain::in_memory::SharedMemory>,
+    receipt_memory_reader: moved_blockchain::receipt::ReceiptMemoryReader,
+    receipt_memory: Option<moved_blockchain::receipt::ReceiptMemory>,
+    trie_db: Arc<moved_state::InMemoryTrieDb>,
+}
+
+impl InMemoryDependencies {
+    pub fn new() -> Self {
```

Comment:
Now that `InMemoryDependencies` has a `new()` method it should also have

```rust
impl Default for InMemoryDependencies {
    fn default() -> Self {
        Self::new()
    }
}
```

I'm surprised clippy didn't complain about it.